exporter/kinesisexporter/producer.go

```diff
@@ -0,0 +1,81 @@
+package kinesisexporter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	omnition "github.com/signalfx/omnition-kinesis-producer"
+	"github.com/signalfx/omnition-kinesis-producer/loggers/kpzap"
+	"go.uber.org/zap"
+	"time"
+)
+
+// producer provides the interface for a kinesis producer. The producer
+// implementation abstracts the interaction with the kinesis producer library
+// used from the exporter
+type producer interface {
```

Comment:
Considering that this looks to be an internal interface, does it need to exist? 

Only things within this package can access it so does it provide much value?

Reply:
It was only added for testing purposes, without the interface we can't mock the producer